src/include/sof/drivers/esai.h

```diff
@@ -38,16 +41,162 @@
 #define REG_ESAI_TCCR           0xD8
 #define REG_ESAI_RCR            0xDC
 #define REG_ESAI_RCCR           0xE0
-#define REG_ESAI_xCR(tx)        (tx ? REG_ESAI_TCR : REG_ESAI_RCR)
-#define REG_ESAI_xCCR(tx)       (tx ? REG_ESAI_TCCR : REG_ESAI_RCCR)
+#define REG_ESAI_xCR(rx)        (rx ? REG_ESAI_RCR : REG_ESAI_TCR)
+#define REG_ESAI_xCCR(rx)       (rx ? REG_ESAI_RCCR : REG_ESAI_TCCR)
 #define REG_ESAI_TSMA           0xE4
 #define REG_ESAI_TSMB           0xE8
 #define REG_ESAI_RSMA           0xEC
 #define REG_ESAI_RSMB           0xF0
-#define REG_ESAI_xSMA(tx)       (tx ? REG_ESAI_TSMA : REG_ESAI_RSMA)
-#define REG_ESAI_xSMB(tx)       (tx ? REG_ESAI_TSMB : REG_ESAI_RSMB)
+#define REG_ESAI_xSMA(rx)       (rx ? REG_ESAI_RSMA : REG_ESAI_TSMA)
+#define REG_ESAI_xSMB(rx)       (rx ? REG_ESAI_RSMB : REG_ESAI_TSMB)
 #define REG_ESAI_PRRC           0xF8
 #define REG_ESAI_PCRC           0xFC
 
+#define ESAI_ECR_ETI		BIT(19)
+#define ESAI_ECR_ETO		BIT(18)
+#define ESAI_ECR_ERI		BIT(17)
+#define ESAI_ECR_ERO		BIT(16)
+#define ESAI_ECR_ERST		BIT(1)
+#define ESAI_ECR_ESAIEN		BIT(0)
+
+#define ESAI_ESR_TINIT		BIT(10)
+#define ESAI_ESR_RFF		BIT(9)
+#define ESAI_ESR_TFE		BIT(8)
+#define ESAI_ESR_TLS		BIT(7)
+#define ESAI_ESR_TDE		BIT(6)
+#define ESAI_ESR_TED		BIT(5)
+#define ESAI_ESR_TD		BIT(4)
+#define ESAI_ESR_RLS		BIT(3)
+#define ESAI_ESR_RDE		BIT(2)
+#define ESAI_ESR_RED		BIT(1)
+#define ESAI_ESR_RD		BIT(0)
+
+#define ESAI_xFCR_TIEN		BIT(19)
+#define ESAI_xFCR_xWA_SHIFT	16
+#define ESAI_xFCR_xWA_WIDTH	3
+#define ESAI_xFCR_xWA_MASK	MASK(18, 16)
+#define ESAI_xFCR_xWA(v)	SET_BITS(18, 16, 8 - ((v) >> 2))
+#define ESAI_xFCR_xFWM_SHIFT	8
+#define ESAI_xFCR_xFWM_WIDTH	8
+#define ESAI_xFCR_xFWM_MASK	MASK(15, 8)
+#define ESAI_xFCR_xFWM(v)	SET_BITS(15, 8, (v) - 1)
+#define ESAI_xFCR_xE_SHIFT	2
+#define ESAI_xFCR_TE_WIDTH	6
+#define ESAI_xFCR_RE_WIDTH	4
+#define ESAI_xFCR_TE_MASK	MASK(7, 2)
+#define ESAI_xFCR_RE_MASK	MASK(5, 2)
+#define ESAI_xFCR_xE_MASK(rx)	(rx ? ESAI_xFCR_RE_MASK : ESAI_xFCR_TE_MASK)
+#define ESAI_xFCR_TE(x)		SET_BITS(7, 2, MASK((x) - 1, 0))
+#define ESAI_xFCR_RE(x)		SET_BITS(5, 2, MASK((x) - 1, 0))
+#define ESAI_xFCR_xE(rx, x)	(rx ? ESAI_xFCR_RE(x) : ESAI_xFCR_TE(x))
+#define ESAI_xFCR_xFR_SHIFT	1
+#define ESAI_xFCR_xFR_MASK	BIT(1)
+#define ESAI_xFCR_xFR		BIT(1)
+#define ESAI_xFCR_xFEN_SHIFT	0
+#define ESAI_xFCR_xFEN		BIT(0)
+#define ESAI_xFCR_xFEN_MASK	BIT(0)
+
+#define ESAI_xFSR_NTFO_MASK	MASK(14, 12)
+#define ESAI_xFSR_NTFI_MASK	MASK(10, 8)
+#define ESAI_xFSR_NRFO_MASK	MASK(9, 8)
+#define ESAI_xFSR_NRFI_MASK	MASK(13, 12)
+#define ESAI_xFSR_xFCNT_MASK	MASK(7, 0)
+
+#define ESAI_SAISR_TODFE	BIT(17)
+#define ESAI_SAISR_TEDE		BIT(16)
```

Comment:
might want to align them

Reply:
Oops, will do

Reply:
That looks like an indent too much due to Github and being very close to a multiple of 8 before the tabs. So it's Github indent that is broken :)